### PR TITLE
feat(retries): Use config params for retry parameters

### PR DESCRIPTION
**Pull request by PatStiles**
# Use config file for retry parameters

## Description

Use config file for retry parameters

## Type of change

Please delete options that are not relevant.

- [x] New feature

### CODE REVIEW

File: core/chainio/avs_writer.go [hunk 1]
```diff
@@ -104,12 +112,24 @@ func (w *AvsWriter) SendAggregatedResponse(batchIdentifierHash [32]byte, batchMe
 	txOpts.NoSend = false
 	i := 0
 
+	// Set Retry config for RespondToTaskV2
+	respondToTaskV2Config := retry.EthCallRetryParams()
+	respondToTaskV2Config.NumRetries = respondToTaskV2NumRetries
```
Comment (uri-99) [inline core/chainio/avs_writer.go]: should be chain because its a send

File: core/chainio/avs_writer.go [hunk 2]
```diff
@@ -104,12 +112,24 @@ func (w *AvsWriter) SendAggregatedResponse(batchIdentifierHash [32]byte, batchMe
...
+	respondToTaskV2Config.MaxInterval = respondToTaskV2MaxInterval
+	respondToTaskV2Config.MaxElapsedTime = respondToTaskV2MaxElapsedTime
+
+	// Set Retry config for WaitForTxRetryable
+	waitForTxConfig := retry.EthCallRetryParams()
+	waitForTxConfig.MaxInterval = waitForTxMaxInterval
```
Comment (uri-99) [inline core/chainio/avs_writer.go]: not sure if network or chain

File: core/chainio/avs_writer.go [hunk 3]
```diff
@@ -92,7 +92,7 @@ func NewAvsWriterFromConfig(baseConfig *config.BaseConfig, ecdsaConfig *config.E
 func (w *AvsWriter) SendAggregatedResponse(batchIdentifierHash [32]byte, batchMerkleRoot [32]byte, senderAddress [20]byte, nonSignerStakesAndSignature servicemanager.IBLSSignatureCheckerNonSignerStakesAndSignature, gasBumpPercentage uint, gasBumpIncrementalPercentage uint, timeToWaitBeforeBump time.Duration, onGasPriceBumped func(*big.Int)) (*types.Receipt, error) {
 	txOpts := *w.Signer.GetTxOpts()
 	txOpts.NoSend = true // simulate the transaction
-	simTx, err := w.RespondToTaskV2Retryable(&txOpts, batchMerkleRoot, senderAddress, nonSignerStakesAndSignature)
+	simTx, err := w.RespondToTaskV2Retryable(&txOpts, batchMerkleRoot, senderAddress, nonSignerStakesAndSignature, retry.SendToChainRetryParams())
```
Comment (uri-99) [inline core/chainio/avs_writer.go:95]: la pregunta del millón

File: core/chainio/avs_writer.go [hunk 4]
```diff
@@ -144,7 +148,7 @@ func (w *AvsWriter) SendAggregatedResponse(batchIdentifierHash [32]byte, batchMe
 				}
 			}
 			w.logger.Infof("Receipts for old transactions not found, will check if the batch state has been responded", "merkle root", batchMerkleRootHashString)
-			batchState, _ := w.BatchesStateRetryable(&bind.CallOpts{}, batchIdentifierHash)
+			batchState, _ := w.BatchesStateRetryable(&bind.CallOpts{}, batchIdentifierHash, retry.SendToChainRetryParams())
```
Comment (uri-99) [inline core/chainio/avs_writer.go]: i think this is network

File: core/retry_test.go [hunk 5]
```diff
@@ -153,7 +169,7 @@ func TestWaitForTransactionReceipt(t *testing.T) {
 	}
 
 	// Assert Call succeeds when Anvil running
-	_, err = utils.WaitForTransactionReceiptRetryable(*client, *client, hash, time.Second*45)
+	_, err = utils.WaitForTransactionReceiptRetryable(*client, *client, hash, retry.NetworkRetryParams())
```
Comment (uri-99) [inline core/retry_test.go]: chain

File: core/retry_test.go [hunk 6]
```diff
@@ -165,7 +181,7 @@ func TestWaitForTransactionReceipt(t *testing.T) {
 		return
 	}
 
-	_, err = utils.WaitForTransactionReceiptRetryable(*client, *client, hash, time.Second*45)
+	_, err = utils.WaitForTransactionReceiptRetryable(*client, *client, hash, retry.NetworkRetryParams())
```
Comment (uri-99) [inline core/retry_test.go]: chain

File: core/retry_test.go [hunk 7]
```diff
@@ -181,7 +197,7 @@ func TestWaitForTransactionReceipt(t *testing.T) {
 		t.Errorf("Error setting up Anvil: %s\n", err)
 	}
 
-	_, err = utils.WaitForTransactionReceiptRetryable(*client, *client, hash, time.Second*45)
+	_, err = utils.WaitForTransactionReceiptRetryable(*client, *client, hash, retry.NetworkRetryParams())
```
Comment (uri-99) [inline core/retry_test.go]: chain

File: core/retry_test.go [hunk 8]
```diff
@@ -646,7 +662,7 @@ func TestRespondToTaskV2(t *testing.T) {
 	zero_bytes := [32]byte{}
 
 	// NOTE: With zero bytes the tx reverts
-	_, err = w.RespondToTaskV2Retryable(&txOpts, zero_bytes, aggregator_address, nonSignerStakesAndSignature)
+	_, err = w.RespondToTaskV2Retryable(&txOpts, zero_bytes, aggregator_address, nonSignerStakesAndSignature, retry.NetworkRetryParams())
```
Comment (uri-99) [inline core/retry_test.go]: chain

File: core/retry_test.go [hunk 9]
```diff
@@ -656,7 +672,7 @@ func TestRespondToTaskV2(t *testing.T) {
 		t.Errorf("Error killing process: %v\n", err)
 	}
 
-	_, err = w.RespondToTaskV2Retryable(&txOpts, zero_bytes, aggregator_address, nonSignerStakesAndSignature)
+	_, err = w.RespondToTaskV2Retryable(&txOpts, zero_bytes, aggregator_address, nonSignerStakesAndSignature, retry.NetworkRetryParams())
```
Comment (uri-99) [inline core/retry_test.go]: chain

File: core/retry_test.go [hunk 10]
```diff
@@ -671,7 +687,7 @@ func TestRespondToTaskV2(t *testing.T) {
 	}
 
 	// NOTE: With zero bytes the tx reverts
-	_, err = w.RespondToTaskV2Retryable(&txOpts, zero_bytes, aggregator_address, nonSignerStakesAndSignature)
+	_, err = w.RespondToTaskV2Retryable(&txOpts, zero_bytes, aggregator_address, nonSignerStakesAndSignature, retry.NetworkRetryParams())
```
Comment (uri-99) [inline core/retry_test.go]: chain

**Review by uri-99 — CHANGES_REQUESTED** [review]

File: core/chainio/avs_writer.go (hunk 3 above)
Comment (JuArce) [reply core/chainio/avs_writer.go:95]: Yes

File: core/retry_test.go (hunk 5 above)
Comment (JuArce) [reply core/retry_test.go]: agree

File: core/retry_test.go (hunk 6 above)
Comment (JuArce) [reply core/retry_test.go]: agree

File: core/retry_test.go (hunk 7 above)
Comment (JuArce) [reply core/retry_test.go]: agree

File: core/retry_test.go (hunk 8 above)
Comment (JuArce) [reply core/retry_test.go]: agree

**Review by uri-99 — APPROVED** [review]

**Review by uri-99 — APPROVED** [review]

**Review by MauroToscano — APPROVED** [review]

**Review by JuArce — APPROVED** [review]

**Pull request merged**